v2/internal/loader/writing_loader.go

```diff
@@ -50,17 +51,31 @@ type writingReader struct {
 }
 
 func (w *writingReader) Read(p []byte) (int, error) {
+	buf := bytes.NewBuffer(nil)
```

Comment:
Is it possible to avoid creating a new buffer here unless the skip falls within this block?

Reply:
the complexity of getting the potential partial-write edge cases of each of the 3 internal writes worries me a little bit, but yeah i'll take a look